packages/shared/src/lib/core/nfts/actions/checkForUntrackedNfts.ts

```diff
@@ -27,23 +27,27 @@ export function checkForUntrackedNfts(account: IAccountState): void {
 
         const explorerNfts = await explorerApi.getAssetsForAddress(evmAddress, NftStandard.Erc721)
         for (const explorerNft of explorerNfts) {
-            const { token, value } = explorerNft
-            const { address } = token
-            const contract = chain.getContract(ContractType.Erc721, address)
+            try {
+                const { token, value } = explorerNft
+                const { address } = token
+                const contract = chain.getContract(ContractType.Erc721, address)
 
-            const isEnumerable = await contract.methods.supportsInterface(Erc721InterfaceId.Enumerable).call()
-            if (isEnumerable) {
-                await Promise.all(
-                    Array.from({ length: Number(value) }).map(async (_, idx) => {
-                        const tokenId = await contract.methods.tokenOfOwnerByIndex(evmAddress, idx).call()
-                        await persistNft(token, tokenId, contract)
-                    })
-                )
-            } else {
-                await persistNft(token, DEFAULT_NFT_TOKEN_ID, contract)
-            }
+                const isEnumerable = await contract.methods.supportsInterface(Erc721InterfaceId.Enumerable).call()
+                if (isEnumerable) {
+                    await Promise.all(
+                        Array.from({ length: Number(value) }).map(async (_, idx) => {
+                            const tokenId = await contract.methods.tokenOfOwnerByIndex(evmAddress, idx).call()
+                            await persistNft(token, tokenId, contract)
+                        })
+                    )
+                } else {
+                    await persistNft(token, DEFAULT_NFT_TOKEN_ID, contract)
+                }
 
-            addNewTrackedNftToActiveProfile(networkId, address, TokenTrackingStatus.AutomaticallyTracked)
+                addNewTrackedNftToActiveProfile(networkId, address, TokenTrackingStatus.AutomaticallyTracked)
+            } catch (err) {
+                throw new Error(`Unable to persist NFT with address ${explorerNft.token.address}`)
```

Comment:
This error needs to be handled in `pollLayer2BalanceForAccount()`, so that it doesn't result in a runtime crash.

Reply:
![image](https://github.com/bloomwalletio/bloom/assets/45079109/1e45345a-8742-46ab-875b-71f6142d3476)